### ISSUE TITLE
'A' in the taxid from standard classification and extra space in k-mer profile column

### ISSUE THREAD

**Issue by Yiheng323** [DerrickWood/kraken2#126]
Hi all,
I was trying to analyze the k-mer profile column of the output file and found two small issues which I am not sure where are they from so just put a note here.

1. In the output of standard_database search, I found some k-mers hit "A", which a particular profile looks like "A:100". Maybe because I build the database wrongly but couldn't figure out the reason for this.
2. I was trying to read the k-mer profile column using python by splitting the column with space. I found some cells in that column contain an excessive space at the end, so it generate empty element and when I try to further split each k-mer profile (eg. 9606:400) by ":" it returned an error. I am also not sure how this happened and it definitely can be from python pandas module which I used for reading the output file.

The first issue wasn't important to me and I also fixed the second issue. My goal basically is to generate a bed file from the k-mer profile so if anyone knows any useful tool it would be nice. I am also happy to share my code for doing that if anyone wants.

Thanks,
Yiheng

**Comment by jenniferlu717** [comment]
1. The kmer assignment to "A" occurs anytime sample sequences contain a non-standard base (such as N)
2. I'm not sure what you were doing exactly in python, but when reading in any line: line.strip() will remove the end line ("\n") character and then you can split the columns by tabs. 

I also don't know why you want to make a bed file from the k-mer profile but I do not have a tool for that.

**Comment by Yiheng323** [comment]
Hi Jennifer,
Thank you very much for your prompt reply.
I was following your 'Removing contaminants from databases' paper and try to mask those regions from a custom fungal dataset that hit the standard kraken database (except fungi). The only tool I know was bedtools maskfasta so I need to generate the bed file from the kraken2 output. I think I sent you an email before to ask how to do that a while ago...
I used pandas.read_csv to separate the columns first, and the ("\n") only occurred in few of lines (in my case, 63/78k classified reads) which I am not sure why... I used filter() function to remove the empty element after I separate the k-mer profile by space, but I think strip() function will work too and should be faster. Anyways it's just a very minor thing and not difficult to solve, and the classification worked great and super fast in all my cases.

Yiheng
